src/Components/ServiceScene/Breakdowns/FieldsAggregatedBreakdownScene.tsx

```diff
@@ -44,59 +53,77 @@ export class FieldsAggregatedBreakdownScene extends SceneObjectBase<FieldsAggreg
     this.addActivationHandler(this.onActivate.bind(this));
   }
 
-  private onDetectedFieldsChange = (newState: QueryRunnerState, prevState: QueryRunnerState) => {
-    const newNamesField = getDetectedFieldsNamesFromQueryRunnerState(newState);
-    const prevNamesField = getDetectedFieldsNamesFromQueryRunnerState(prevState);
-
-    if (newState.data?.state === LoadingState.Done && !areArraysEqual(newNamesField?.values, prevNamesField?.values)) {
+  private onDetectedFieldsChange = (newState: QueryRunnerState) => {
+    if (newState.data?.state === LoadingState.Done) {
       //@todo cardinality looks wrong in API response
-      const cardinalityMap = this.calculateCardinalityMap(newState);
-
-      // Iterate through all the layouts
-      this.state.body?.state.layouts.forEach((layoutObj) => {
-        const layout = layoutObj as SceneCSSGridLayout;
-        // populate set of new list of fields
-        const newFieldsSet = new Set<string>(newNamesField?.values);
-        const updatedChildren = layout.state.children as SceneCSSGridItem[];
-
-        // Iterate through all the existing panels
-        for (let i = 0; i < updatedChildren.length; i++) {
-          const gridItem = layout.state.children[i] as SceneCSSGridItem;
-          const panel = gridItem.state.body as VizPanel;
-
-          if (newFieldsSet.has(panel.state.title)) {
-            // If the new response has this field, delete it from the set, but leave it in the layout
-            newFieldsSet.delete(panel.state.title);
-          } else {
-            // Otherwise if the panel doesn't exist in the response, delete it from the layout
-            updatedChildren.splice(i, 1);
-            // And make sure to update the index, or we'll skip the next one
-            i--;
+      this.updateChildren(newState);
+    }
+  };
+
+  private updateChildren(newState: QueryRunnerState, newParser: ParserType | undefined = undefined) {
+    const detectedFieldsFrame = getDetectedFieldsFrameFromQueryRunnerState(newState);
+    const newNamesField = getDetectedFieldsNamesFromQueryRunnerState(newState);
+    const newParsersField = getDetectedFieldsParsersFromQueryRunnerState(newState);
+    const cardinalityMap = this.calculateCardinalityMap(newState);
+
+    // Iterate through all the layouts
+    this.state.body?.state.layouts.forEach((layoutObj) => {
+      const layout = layoutObj as SceneCSSGridLayout;
+      // populate set of new list of fields
+      const newFieldsSet = new Set<string>(newNamesField?.values);
+      const updatedChildren = layout.state.children as SceneCSSGridItem[];
+
+      // Iterate through all the existing panels
+      for (let i = 0; i < updatedChildren.length; i++) {
+        const gridItem = layout.state.children[i] as SceneCSSGridItem;
+        const panel = gridItem.state.body as VizPanel;
+
+        if (newParser) {
```

Comment:
This fixes a bug in the field names scene where adding a field filter with a different parser then the panel requires i.e. a metadata field panel breaks when a logfmt/json field include/exclude/numeric filter is added.